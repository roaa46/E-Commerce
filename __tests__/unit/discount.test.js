const { showDiscount, calculateNewPrice, getDiscountAmount } = require('../../helpers/discount');

describe('showDiscount', () => {
  test('should return true when discount is greater than 0', () => {
    expect(showDiscount(10)).toBe(true);
  });

  test('should return false when discount is 0', () => {
    expect(showDiscount(0)).toBe(false);
  });

  test('should return false when discount is negative', () => {
    expect(showDiscount(-5)).toBe(false);
  });


});

describe('calculateNewPrice', () => {
  test('should return "90.00" when price is 100 and discount is 10', () => {
    expect(calculateNewPrice(100, 10)).toBe("90.00");
  });

  test('should return "0.00" when price is 0', () => {
    expect(calculateNewPrice(0, 50)).toBe("0.00");
  });

  test('should return "100.00" when discount is 0', () => {
    expect(calculateNewPrice(100, 0)).toBe("100.00");
  });
});

describe('calculateNewPrice with negative price', () => {
  
  test('should return "0.00" when price is negative', () => {
    expect(() => getDiscountAmount(-100, 10)).toThrowErrorMatchingSnapshot();
  });

  test('should return 0 when price is 0 regardless of discount', () => {
      expect(getDiscountAmount(0, 50)).toBe(0);
    });
});
