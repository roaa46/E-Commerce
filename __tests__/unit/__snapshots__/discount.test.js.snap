// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateNewPrice with negative price should return "0.00" when price is negative 1`] = `"Price cannot be negative"`;

exports[`showDiscount should throw matching error snapshot 1`] = `"getDiscountAmount is not defined"`;
